src/iuse.cpp

```diff
@@ -2747,160 +2747,542 @@ int iuse::sew(player *p, item *it, bool, point)
+                if (p->dex_cur < 8 && one_in(p->dex_cur)) {
+                    rn -= rng(2, 6);
+                }
+                if (p->dex_cur >= 16 || (p->dex_cur > 8 && one_in(16 - p->dex_cur))) {
+                    rn += rng(2, 6);
+                }
+                if (p->dex_cur > 16) {
+                    rn += rng(0, p->dex_cur - 16);
+                }
+                if (rn <= 4) {
+                    p->add_msg_if_player(m_bad, _("You damage your %s!"), fix->tname().c_str());
+                    fix->damage++;
+                } else
+                    if (rn >= 12 && fix->has_flag("VARSIZE") && !fix->has_flag("FIT")) {
+                        p->add_msg_if_player(m_good, _("You take your %s in, improving the fit."), fix->tname().c_str());
+                        fix->item_tags.insert("FIT");
+                    } else
+                        if (rn >= 12 && (fix->has_flag("FIT") || !fix->has_flag("VARSIZE"))) {
+                            p->add_msg_if_player(m_neutral, _("Your %s is already fully repaired."), fix->tname().c_str());
+                            return 0;
+                        } else {
+                            p->add_msg_if_player(m_neutral, _("You practice your sewing."));
+                        }
+            } else {
+                p->add_msg_if_player(m_info, _("Your %s is already enhanced."), fix->tname().c_str());
+                return 0;
+            }
+
+        return thread_used;
+
     }
+    case 2: {
+        int thread_used = 5;
+
+        int pos = g->inv_for_filter( _("Enhance what?"), []( const item & itm ) {
+            return itm.made_of( "cotton" ) ||
+                   itm.made_of( "leather" ) ||
+                   itm.made_of( "fur" ) ||
+                   itm.made_of( "nomex" ) ||
+                   itm.made_of( "wool" );
+        } );
+        item *mod = &(p->i_at(pos));
+        if (mod == NULL || mod->is_null()) {
+            p->add_msg_if_player(m_info, _("You do not have that item!"));
+            return 0;
+        };
+        itype_id repair_item = "none";
+        std::vector<std::string> plurals;
+        std::vector<itype_id> repair_items;
+        std::string plural = "";
+        if( mod == it
+                || std::find(repair_items.begin(), repair_items.end(), mod->typeId()) != repair_items.end()) {
+            p->add_msg_if_player(m_info, _("This can be used to repair other items, not itself."));
+            return 0;
+        };
+        if(((mod->furred == 1) && (mod->pocketed == 1)) || ((mod->furred == 1) && (mod->leather_padded == 1)) ||
+                ((mod->pocketed == 1) && (mod->leather_padded == 1)) || ((mod->furred == 1)
+                        && (mod->kevlar_padded == 1)) ||
+                ((mod->kevlar_padded == 1) && (mod->pocketed == 1)) || ((mod->leather_padded == 1)
+                        && (mod->kevlar_padded == 1))) { //This is a mess. Have to block every combo of 2. Probably a better way.
```

Comment:
Bitflag, if possible, and then if it's not a power of two, it's got a second modification.